app/controllers/concerns/api_filtering.rb

```diff
@@ -8,11 +8,17 @@ module ApiFiltering
   included do
     class_eval do
       class_attribute :allowed_filter_names
+      class_attribute :allowed_filter_types
     end
   end
 
   class UnknownFilterError < NoMethodError
   end
+  class BadFilterValueError < HalApi::Errors::ApiError
+    def initialize(msg)
+      super(msg, 400)
```

Comment:
I wanted these to be an HTTP 400, which is already built into the HalApi error handling.